File: src/main/java/com/duff/api/domain/Beer.java
package com.duff.api.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;

import javax.persistence.*;

@Data
@Entity
@Table(name = "beer")
public class Beer {

    @Id
    @Column(name = "style")
    private String style;

    @Column(name = "min_temperature")
    private double minTemperature;

    @Column(name = "max_temperature")
    private double maxTemperature;

    @JsonIgnore
    public double getAverage() {
        return (maxTemperature + minTemperature) / 2;
    }
}

File: src/domain/mongodb/schemas/Beer.ts
import mongoose from "mongoose";

const beerSchema = new mongoose.Schema({
  minTemperature: {
    type: Number,
    required: true
  },
  maxTemperature: {
    type: Number,
    required: true
  },
  style: {
    type: String,
    required: true
  }
});

export const Beer = mongoose.model('Beer', beerSchema);